examples/plot_spm_auditory.py

```diff
@@ -42,29 +43,29 @@
 frametimes = np.linspace(0, (nscans - 1) * tr, nscans)
 drift_model = 'Cosine'
 hrf_model = 'Canonical With Derivative'
-hfcut = 2 * 2 * epoch_duration
+period_cut = 2 * 2 * epoch_duration
 design_matrix = make_design_matrix(
     frametimes, paradigm, hrf_model=hrf_model, drift_model=drift_model,
-    hfcut=hfcut)
+    period_cut=period_cut)
 
 # plot and save design matrix
-ax = design_matrix.show()
+ax = plot_design_matrix(design_matrix)
 ax.set_position([.05, .25, .9, .65])
 ax.set_title('Design matrix')
 
 # specify contrasts
 contrasts = {}
-n_columns = len(design_matrix.names)
-for i in range(n_conditions):
-    contrasts['%s' % design_matrix.names[2 * i]] = np.eye(n_columns)[2 * i]
+_, matrix, names = check_design_matrix(design_matrix)
+for i in range(len(names)):
+    contrasts['%s' % names[i]] = np.eye(len(names))[i]
```

Comment:
Hmmm, I don't know enough about the context to know whether the '2 * i -> 'i' replacement was done on purpose. Also why was it done in this example and not in the plot_localizer_analysis.py example?

Otherwise, same comment as above: you can take the matrix creation out of the loop and use `names[i]` rather than `'%s' % names[i]`.